pkg/apis/extensions/v1alpha1/helper/helper.go

```diff
@@ -31,10 +31,13 @@ func ClusterAutoscalerRequired(pools []extensionsv1alpha1.WorkerPool) bool {
 	return false
 }
 
-// GetDNSRecordType returns the appropriate DNS record type (A or CNAME) for the given address.
+// GetDNSRecordType returns the appropriate DNS record type (A/AAAA or CNAME) for the given address.
 func GetDNSRecordType(address string) extensionsv1alpha1.DNSRecordType {
```

Comment:
Can you adapt provider-local's ingress controller to use this function for creating the matching `recordType`?
https://github.com/gardener/gardener/blob/cffdb9fd10d598be614d90055ef1bf79d4e5a791/pkg/provider-local/controller/ingress/reconciler.go#L116